frontend/views/components/ContributionItem.vue

```diff
@@ -0,0 +1,160 @@
+<template lang='pug'>
+.c-contribution-item
+  i(:class='iconClass')
+
+  div(v-if='hasWhoElse')
+    transition(name='replacelist')
+      div(
+        v-if='isVisible'
+        key='visible'
+      )
+        .c-contribution-list(v-html='listOfName')
+
+        i18n.is-unstyled.is-link-inherit.link(
+          tag='button'
+          type='button'
+          @click='isVisible = !isVisible'
+        ) Hide
+
+      div(
+        v-else
+        key='hidden'
+        @click='isVisible = !isVisible'
+        v-html='contributionText'
+      )
+
+  .c-contribution-list(
+    v-else=''
+    v-html='contributionText'
+  )
+</template>
+
+<script>
+import L from '@view-utils/translations.js'
+
+export default {
+  name: 'ContributionItem',
+  props: {
+    who: [String, Array],
+    what: String,
+    action: {
+      type: String,
+      validator: function (value) {
+        // The value must match one of these strings
+        return ['RECEIVING', 'GIVING'].indexOf(value) !== -1
+      },
+      default: 'RECEIVING'
+    },
+    type: {
+      type: String,
+      validator: function (value) {
+        // The value must match one of these strings
+        return ['NON_MONETARY', 'MONETARY'].indexOf(value) !== -1
+      },
+      default: 'NON_MONETARY'
+    }
+  },
+  data () {
+    return {
+      isVisible: false
+    }
+  },
+  computed: {
+    contributionText () {
+      if (this.hasWhoElse) {
+        const html = {
+          service: `<span class="has-text-bold">${this.what}</span>`,
+          numMembers: `<button class="is-unstyled is-link-inherit link">${this.otherContributor}`,
+          _button: '</button>'
+        }
+
+        if (this.action === 'RECEIVING' && this.type === 'MONETARY') {
+          return L('{service} from {numMembers} members{_button}', html)
+        } else {
+          return L('A total of {service} to {numMembers} members{_button}', html)
+        }
```

Comment:
To make this a bit clearer, we should separate out `numMembers` from the button markup, so that it's clear where the opening tag for `{_button}` is. Something like:

```js
        const html = {
          service: `<span class="has-text-bold">${this.what}</span>`,
          button_: '<button class="is-unstyled is-link-inherit link">',
          numMembers: this.otherContributor,
          _button: '</button>'
        }
```

Now we can write:

```js
return L('{service} from {button_}{numMembers} members{_button}', html)
```

etc.